extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -137,6 +138,16 @@ public void build(NarayanaJtaRecorder recorder,
         builder.addBeanClass(TransactionalInterceptorNotSupported.class);
         additionalBeans.produce(builder.build());
 
+        // This must be done before setNodeName as the code in setNodeName will create a TSM based on the value of this property
+        recorder.disableTransactionStatusManager();
+        recorder.setNodeName(transactions);
```

Comment:
Did someone check if we end up with having the same node names across all instances?

Reply:
I don't think Amos actually changed these lines, the diff is showing because the other lines were reordered. I believe that the only change is to move the code that sets up the defaultProperties into a @Record(STATIC_INIT) build step (he added the new method `public void setProperties(NarayanaJtaRecorder recorder)`. Is that correct @zhfeng ?

Reply:
Yeah, it is. Thanks @mmusgrov !

Reply:
@Sanne I don't understand here. What are you mean all intances?

Reply:
By "instances" I mean instances of the application. If I start 10 copies of the same Quarkus application, it's my understanding that it's essential that they are assigned a different Node Name - or at least to offer the possibility of users to configure different node names.

By seeing the `setNodeName` invoked on a build-time recorder, it makes me worry that this configuration setting wouldn't be honoursed at runtime: I'm suggesting to please check this as it would be a grave mistake.

Reply:
Ok, thanks.